app/uk/gov/hmrc/eusubsidycompliancefrontend/services/EmailService.scala

```diff
@@ -107,6 +109,22 @@ class EmailService @Inject() (
         }
     }
 
+  def hasVerifiedEmail(eori: EORI)(implicit
+    hc: HeaderCarrier,
+    ec: ExecutionContext
+  ): Future[Option[VerifiedStatus]] =
+    verifiedEoriCache.get(eori).flatMap {
+      case Some(_) => Future.successful(Some(Verified))
+      case None =>
+        for {
+          resp <- retrieveEmailByEORI(eori).flatMap {
+            case RetrieveEmailResponse(EmailType.VerifiedEmail, _) =>
+              verifiedEoriCache.put(VerifiedEori(eori)).map { _ => Some(Verified) }
```

Comment:
Just for the purpose of knowledge sharing, cats has this `as` method to make this look cleaner

```suggestion
              verifiedEoriCache.put(VerifiedEori(eori)).as(Some(Verified))
```